### PR TITLE
Bugfix/lock state spam

### PR DESCRIPTION
**Pull request by allexzander**

### CODE REVIEW

File: src/gui/folderwatcher.cpp [hunk 1]
```diff
@@ -157,6 +160,20 @@ void FolderWatcher::startNotificationTestWhenReady()
     });
 }
 
+void FolderWatcher::lockChageDebouncingTimerTimedOut()
```
Comment (mgallien) [inline src/gui/folderwatcher.cpp:163]: you have a typo there

File: src/libsync/account.cpp [hunk 2]
```diff
@@ -924,14 +935,24 @@ std::shared_ptr<UserStatusConnector> Account::userStatusConnector() const
...
 {
-    auto job = std::make_unique<LockFileJob>(sharedFromThis(), journal, serverRelativePath, lockStatus);
-    connect(job.get(), &LockFileJob::finishedWithoutError, this, [this]() {
+    auto& lockStatusJobInProgress = _lockStatusChangeInprogress[serverRelativePath];
+    if (lockStatusJobInProgress.contains(lockStatus)) {
+        qCWarning(lcAccount) << "Already running a job with lockStatus:" << lockStatus << " for: " << serverRelativePath;
+        return;
+    }
+    lockStatusJobInProgress.push_back(lockStatus);
+    auto job = std::make_unique<LockFileJob>(sharedFromThis(), journal, serverRelativePath, remoteSyncPathWithTrailingSlash, localSyncPath, lockStatus);
```
Comment (mgallien) [inline src/libsync/account.cpp:949]: will not this code delete the job instance too early ?

**Review by mgallien — CHANGES_REQUESTED** [review]

File: src/gui/folder.cpp [hunk 3]
```diff
@@ -690,10 +694,7 @@ void Folder::slotLockedFilesFound(const QSet<QString> &files)
         const auto fileRecordPath = fileFromLocalPath(file);
         SyncJournalFileRecord rec;
 
-        const auto canLockFile = journalDb()->getFileRecord(fileRecordPath, &rec) && rec.isValid()
-            && (!rec._lockstate._locked
-                || (rec._lockstate._lockOwnerType == static_cast<qint64>(SyncFileItem::LockOwnerType::UserLock)
-                    && rec._lockstate._lockOwnerId == _accountState->account()->davUser()));
+        const auto canLockFile = journalDb()->getFileRecord(fileRecordPath, &rec) && rec.isValid() && !rec._lockstate._locked;
```
[the remark this answers is not in the document]
Comment (allexzander) [reply src/gui/folder.cpp:697]: @mgallien when we already have a lock we should not add another one but lift the previous one first

File: src/libsync/account.cpp (hunk 2 above)
Comment (allexzander) [reply src/libsync/account.cpp:949]: @mgallien not really, see the https://en.cppreference.com/w/cpp/memory/unique_ptr/release which gets called at the end, ownership is being released, but, if you ask me, I would have implemented this without smart pointers, but the code was here before so I did not have time/see reasons to refactor

File: src/gui/folderwatcher.cpp (hunk 1 above)
Comment (allexzander) [reply src/gui/folderwatcher.cpp:163]: @mgallien thanks, fixed

File: src/gui/folder.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (allexzander) [reply src/gui/folder.cpp:697]: @mgallien or should we allow locking already locked? does not make much sense, but I may be wrong

File: src/libsync/account.cpp (hunk 2 above)
Comment (mgallien) [reply src/libsync/account.cpp:949]: missed it during the review

File: src/gui/folder.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (mgallien) [reply src/gui/folder.cpp:697]: @allexzander  so the reply to my question is that before we were allowing lock of an already locked file ?

[the remark this answers is not in the document]
Comment (allexzander) [reply src/gui/folder.cpp:697]: @mgallien yes, incase the existing lock was a user lock and the user id was the current user

**Review by mgallien — APPROVED** [review]

**Pull request merged**